Code/Tools/SceneAPI/SceneCore/Containers/Scene.cpp

```diff
@@ -148,6 +148,7 @@ namespace AZ
                         ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
                         ->Attribute(AZ::Script::Attributes::Module, "scene")
                         ->Constructor<const AZStd::string&>()
+                            ->Attribute(AZ::Script::Attributes::DefaultConstructorOverrideIndex, 0)
```

Comment:
What does this do?

Reply:
I can see the comment but which constructor is being defaulted to here?

```cpp
///< Use a different class constructor as the default constructor in Lua
```

Reply:
It's setting the `Constructor<const AZStd::string&>()` to be the default constructor for lua scripts. This is what was already happening previously, but when I added the default Scene() constructor, that one was becoming the default, and the other constructor wasn't getting used. This attribute sets it back to the way it was, where the string constructor is the one that gets used by default.

There's something a little weird here IMO, since I wouldn't think that adding a non-reflected constructor would change the behavior of the reflected properties, but it does. :( I also don't know quite why it can't pick the correct constructor if it isn't the default, but it couldn't. :(